.travis.yml

```diff
@@ -1,50 +1,120 @@
+            - libnetcdf-dev
+            - libgdal1-dev
+            - libfftw3-dev
+            - libpcre3-dev
+            - liblapack-dev
+            - ghostscript
+            - curl
+            - graphicsmagick
+            - python
+            - python-pip
+    homebrew:
+        packages:
+            - cmake
+            - curl
+            - netcdf
+            - gdal
+            - fftw
+            - pcre2
+            - ghostscript
+            - curl
+            - graphicsmagick
+            - python
 
+# Define environment variables and set build actions
 env:
     global:
         - INSTALLDIR="$HOME/gmt-install-dir"
         - COASTLINEDIR="$HOME/gmt-install-dir/coast"
         - PATH="$INSTALLDIR/bin:$PATH"
         - LD_LIBRARY_PATH="$INSTALLDIR/lib:$LD_LIBRARY_PATH"
-        - if [ "$TRAVIS_EVENT_TYPE" == "cron" ]; then
-            export PYTHON=3.6
-            export CONDA_INSTALL_EXTRA="sphinx"
-            export COVERAGE=true
-          fi
+        - COVERAGE=true
+        - BUILD_DOCS=false
+        - TEST=false
 
+# Create the build matrix with different jobs for each platform and cron jobs
+matrix:
+    include:
+        - name: "Linux"
+          os: linux
+          if: type != cron
+        - name: "Linux"
+          os: linux
+          #if: type = cron
+          env:
+            - TEST="all"
+        - name: "Mac"
+          os: osx
+          if: type != cron
+        - name: "Max"
+          os: osx
+          #if: type = cron
+          env:
+            - TEST="all"
+        # Only build the docs on Linux cron jobs
+        - name: "Linux - Build docs"
+          os: linux
+          #if: type = cron
+          env:
+            - BUILD_DOCS=true
+
+# Setup the build environment
 before_install:
-    # Install GMT dependencies
-    - bash ci/travis-setup.sh
+    - mkdir "$INSTALLDIR"
+    - mkdir "$COASTLINEDIR"
+    - bash ci/download-coastlines.sh
     # Install GMT documentation dependencies
-    - if [ "$TRAVIS_EVENT_TYPE" == "cron" ]; then
-        git clone https://github.com/fatiando/continuous-integration.git;
-        source continuous-integration/travis/setup-miniconda.sh;
+    - if [ "$BUILD_DOCS" == "true" ]; then
+        pip install --user sphinx;
       fi
 
+# Install the package that we want to test
 install:
     # Build and install GMT
-    - bash ci/travis-build.sh;
+    - bash ci/build-gmt.sh;
 
+# Run the actual tests and checks
 script:
+    - cd build
     - gmt defaults -Vd
     - gmt pscoast -R0/10/0/10 -JM6i -Ba -Ggray -P -Vd > test.ps
-    # Build documentations, tests and coverage reports
-    - if [ "$TRAVIS_EVENT_TYPE" == "cron" ]; then
-        cd build; make -j docs_html; cd ..;
-        cd build; make -j check; cd ..;
+    - if [ "$TEST" == "all" ]; then
+        make -j check;
+      fi
+    - if [ "$BUILD_DOCS" == "true" ]; then
+        make -j docs_html;
+      fi
+    - cd ..
+
+# Things to do if the build is successful
+after_success:
```

Comment:
Although some tests fail, we can still upload coverage reports.

Reply:
Right, I forgot about that. At least now the tests really show up as failures instead of passing. Hopefully that red badge will be good motivation to fix them or mark as won't fix :wink: